### ISSUE TITLE
Model License, Evaluation & Training data details

### ISSUE THREAD

**Issue by Benjoyo** [MeetKai/functionary#131]
Thank you for these great models. I have a few questions, if you don’t mind:

1) the code has MIT license but the model weights on HF don’t have any license attached. Do they have the same license? Would you add a license file to HF? Right now the situation is unclear which makes using the models difficult. 

2) did you run any general LLM evaluations to assess possible negative effects of the function calling fine tuning on generalist model capabilities compared to, say, the official instruct fine tune?

3) to the same end, you write that the model is able to chit chat - do you include a fraction of non-function-call chat examples in the training? If so, what is the percentage roughly?

thank you very much in advance :)

**Comment by musabgultekin** [comment]
Hi Bennet!

1. I've added the license to our latest model!
2. We intentionally didn't focus on chat capabilities. The focus was mainly about function calling. And on our internal evaluations, we didn't observe negative effects of function calling to the chat performance. Even for small models, the models can both have function calling accuracy high and have a high quality chat capabilities.
3. Yes on our 2.2 version, we included about 10%~ ultra high quality multi-turn non function-call chat conversations. Which was enough for the model to learn how to chat. Bear in mind that function conversations can also be considered as "normal chats", as they also follow instructions and understand the human intentions.

Because our focus was tool use, If you're heavily depend on chat capabilities, I would suggest you to evaluate manually to see if that works for you. (Since evaluating LLMs are notoriously difficult)
Since we provide the full training script, you can initialize from our fine-tunes to further fine-tune if you want to have custom chat capabilities.

**Comment by Benjoyo** [comment]
Awesome, thank you very much!

I also feel like the models are really good at chatting, but I was worried about catastrophic forgetting of other capabilities if one focuses too much on the function calling part. But I guess if the function calling conversations are diverse enough, that probably won't happen. 
I will test further, but the models seem really good, thank you so much for this work and keep it up! 👍
